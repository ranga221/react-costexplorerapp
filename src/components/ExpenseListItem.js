import React from 'react'
import { connect } from 'react-redux';
import { startRemoveExpense } from '../actions/expenses';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const ExpenseListItem = ({dispatch, id, description, amount, creadtedAt}) => (
<div>
<Link to={`/edit/${id}`}>
<h3>{description}</h3>

</Link>

<p>
{numeral(amount / 100).format('$0,0.00')}
-
{moment(creadtedAt).format('MMMM Do, YYYY')}
</p>
<button onClick={() => {
    dispatch(startRemoveExpense({ id }));
}}>Remove</button>
</div>

);

export default connect()(ExpenseListItem);