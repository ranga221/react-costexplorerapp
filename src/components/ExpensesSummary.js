import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectTotalExpenses from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses';
import { Link } from 'react-router-dom';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
const formatedExpenses = numeral(expensesTotal / 100 ).format('$0,0.00')

    return (
    <div className="page-header">
    <div className="content-container">
    <h1 className="page-header__title">View <span> {expenseCount} </span> {expenseWord} <span>{formatedExpenses}</span></h1>
   <div className="page-header__actions">
   <Link className="button" to="/create"> Add Your Daily Costs</Link>
   </div>
    </div>
    </div>
);
};


const mapStateToProps = (state) => {
const visibleExpenses = selectExpenses(state.expenses, state.filters);

return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectTotalExpenses(visibleExpenses)
};
};

export default connect(mapStateToProps)(ExpensesSummary);