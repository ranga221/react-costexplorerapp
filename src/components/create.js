import React from 'react';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
import {connect} from 'react-redux';

export class addExpensePage extends React.Component{
    onSubmit = (expense) => {
   this.props.startAddExpense(expense);
   this.props.history.push('/dash');
    
};

render() {
return(
    <div>
    <div className="page-header">
    <div className="content-container"></div>
    <h1>Add Expense</h1>
    </div>

    <div className="content-container">  
    <ExpenseForm 
        onSubmit={this.onSubmit} 
    />
    </div>
 
   </div>
);
};
}


const mapDispatchToProps = (dispatch) => ({startAddExpense: (expense) => dispatch(startAddExpense(expense))});

export default connect(undefined, mapDispatchToProps)(addExpensePage);