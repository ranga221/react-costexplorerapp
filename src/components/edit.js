import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses';

const EditExpenseDashboardPage = (props) => {
  
    console.log(props);
  return (
    <div  className="page-header" >
    <div className="content-container"> 
    <ExpenseForm 
    expense={props.expense}
    onSubmit={(expense) => {
      props.dispatch(editExpense(props.match.params.id, expense ));
      props.history.push('/dash');
      
    }}
    />
    </div>
   </div>
  );
};

const mapStateToProps = (state, props) => {
return {
  expense: state.expenses.find((expense) => {
 return expense.id === props.match.params.id;
  })
};

};
export default connect(mapStateToProps)(EditExpenseDashboardPage);