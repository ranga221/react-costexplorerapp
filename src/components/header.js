import React from 'react';
import { BrowserRouter, Route, Switch, Link, NavLink } from 'react-router-dom';

const Header = () => (
    <header>
    <h1>Cost Explorer</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}> Home </NavLink>
    <NavLink to="/create" activeClassName="is-active"> Create New</NavLink>
    <NavLink to="/edit"  activeClassName="is-active"> Edit Existing</NavLink>
    <NavLink to="/help"  activeClassName="is-active"> Help Page</NavLink>
    </header>
);

export default Header;
