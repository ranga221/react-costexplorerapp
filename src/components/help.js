import React from 'react';

const helpExpenseDashboardPage = () => (
    <div>
    This is from my help dashboard components
    </div>
);

export default helpExpenseDashboardPage;