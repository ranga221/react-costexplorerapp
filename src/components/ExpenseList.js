import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
<div className="content-container">
<div className="list-header">
<div>Expense</div>
<div>Amount</div>
</div>

{props.expenses.map((expense)=> {

return <ExpenseListItem key={expense.id}{...expense} />
})}


</div>
);

const mapStateToProps = (state) => {
return{
    expenses: selectExpenses(state.expenses,state.filters)
};
};

export default connect(mapStateToProps)(ExpenseList);