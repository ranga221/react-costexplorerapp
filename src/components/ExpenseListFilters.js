import React from 'react';
import ExpenseList from './ExpenseList';
import { connect } from 'react-redux';
import {setTextFilter, sortByAmount, sortByDate } from '../actions/filters';

const ExpenseListFilters = (props) => (
<div className="conent-container">
<div className="input-group">
<div className="input-group__item">
<input type='text' 
className="text-input"
 value={props.filters.text}  
 onChange={(e) => { 
    props.dispatch(setTextFilter(e.target.value));
}}/>
</div>
<div>
<select
value={props.filters.sortBy}
onChange={(e) =>{
    if(e.target.value === 'date') {
        props.dispatch(sortByDate());
    } else if (e.target.value === 'amount') {
        props.dispatch(sortByAmount());
    }
}}
>
<option value="date">Date</option>
<option value="amount">Amount</option>
</select>
</div>



</div>
<div></div>
</div>


);

const mapStateToProps = (state) => {
return{
    filters: state.filters
}
};

export default connect(mapStateToProps)(ExpenseListFilters);