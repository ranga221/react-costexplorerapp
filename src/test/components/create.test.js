import React from 'react';
import {shallow} from 'enzyme';
import {addExpensePage} from '../../components/create';


test('testing create function', () => {
const onSubmit = jest.fn();
const history = {push: jest.fn()};
const wrapper = shallow(<addExpensePage onSubmit={onSubmit} history={history} />);
expect(wrapper).toMatchSnapshot();
});