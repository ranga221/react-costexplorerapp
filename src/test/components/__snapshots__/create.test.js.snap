// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing create function 1`] = `
<addExpensePage
  history={
    Object {
      "push": [Function],
    }
  }
  onSubmit={[Function]}
/>
`;
